mayastor/src/bdev/nexus/nexus_bdev_children.rs

```diff
@@ -243,6 +244,46 @@ impl Nexus {
         }
     }
 
+    /// Return rebuild task associated with the destination.
+    /// Return error if no rebuild task associated with destination.
+    fn get_rebuild_task(
+        &mut self,
+        destination: &str,
+    ) -> Result<&mut RebuildTask, Error> {
+        match self
+            .rebuilds
+            .iter_mut()
+            .find(|t| t.destination == destination)
+        {
+            Some(rt) => Ok(rt),
+            None => Err(Error::RebuildTaskNotFound {
+                child: destination.to_string(),
+                name: self.name.clone(),
+            }),
+        }
+    }
+
+    /// Stop a rebuild task
+    pub async fn stop_rebuild(
```

Comment:
do we need to make the stop idempotent? eg the stop can take a while if an IO is taking a long time
we can add it as a todo and raise a story

Reply:
As discussed, MOAC should be responsible for the idempotency. We will tell MOAC the current state of the system.